### ISSUE TITLE
Update to Drupal core 10.0

### ISSUE THREAD

**Issue by pyrello** [uiowa/uiowa#6537]
# Motivation

With the impending release of Drupal 10.1, this will be a necessary stop along the way to getting to that version.

# Proposed solution

To check if all our dependencies are compatible with Drupal 10, run the following command:

```
ddev composer prohibits drupal/core 10.0
```

Once all dependencies that need to be upgraded to be compatible and can be upgraded ahead of time have been upgraded, we can run the following command:

```
ddev composer require drupal/core-recommended:^10.0 drupal/core-composer-scaffold:^10.0 drupal/core-project-message:^10.0 --update-with-all-dependencies
```

If there are problems with the above command, it is also possible to update the version constraints for the above packages in composer.json and then run

Remove any patches that have been merged into core, updated any patches that have been rerolled.

```
ddev composer update --lock
```
```[tasklist]
### Tasks
- [ ] https://github.com/uiowa/uiowa/issues/6856
- [ ] https://github.com/uiowa/uiowa/issues/6901
- [ ] https://github.com/uiowa/uiowa/issues/6931
```

**Comment by pyrello** [comment]
This is blocked by #6061.

**Comment by briand44** [comment]
Grooming Conversation.... Probe to understand what all needs to happen to get us to D10. Outline necessary steps so we can tackle them as separate slice(s). Update custom module's `.info.yml` files to be compatible with D9/10 rather than D8/9.

**Comment by joewhitsitt** [comment]
Suggested that [uiowa_auth](https://github.com/uiowa/uiowa_auth) be handled separately so that any changes are complete so that it is clear to others using the module that it is d10 ready. We could have a separate branch for d10 so that d9 sites could still use it.

**Comment by joewhitsitt** [comment]
Forgot that this needs to be upgraded too https://github.com/richardbporter/drush-users-commands/releases/tag/3.2.1

**Comment by pyrello** [comment]
For the `theme_permission` module, it seems that we may have to do some composer trickery to get it to not conflict. I added a patch that makes it compatible with D10 via the module's `.info.yml` file, but that doesn't help us with composer:
```
drupal/theme_permission  2.0.x-dev  requires          drupal/core (^8 || ^9)
```
Another thing we could do is temporarily copy the module into our own codebase until the maintainer releases a D10 compatible version or the module is marked abandoned and we can request to take over its maintainership.

**Comment by pyrello** [comment]
I'm running into a new dependency issue: Drupal 10 requires `psr/log ~3.0.0`.

```
composer prohibits psr/log "~3.0.0"
alchemy/binary-driver   v5.2.0  requires  psr/log (^1.0)         
composer/composer       2.2.22  requires  psr/log (^1.0 || ^2.0)
...
```
```
composer why alchemy/binary-driver
php-ffmpeg/php-ffmpeg v0.19.0 requires alchemy/binary-driver (^1.5 || ~2.0.0 || ^5.0)
```
```
composer why php-ffmpeg/php-ffmpeg 
drupal/media_thumbnails_video 2.0.0 requires php-ffmpeg/php-ffmpeg (>=0.14.0)
```

`drupal/media_thumbnails_video` is on the most recent version :|

**Comment by pyrello** [comment]
Currently stuck on this patch: `Support third party settings for components within a section`.

I haven't found an existing patch that works, so this is another one that may need to be rebased/backported from another version of the patch.

**Comment by pyrello** [comment]
> Currently stuck on this patch: `Support third party settings for components within a section`.

This is resolved, at least as far as having a patch that applies.
